iml-gui/crate/src/components/command_modal.rs

```diff
@@ -880,19 +821,19 @@ impl Model {
                 }
             }
             Select::CommandJobSteps(i, j, ks) => {
-                let a0 = self.commands.iter().find(|x| x.id() == *i);
-                let b0 = self.jobs.iter().find(|x| x.id() == *j);
-                let cs = self.steps.iter().filter(|x| ks.contains(&x.id())).collect::<Vec<_>>();
-                let c0 = if cs.is_empty() { None } else { Some(cs) };
-                match (a0, b0, &c0) {
+                let a0 = self.commands.get(i);
+                let b0 = self.jobs.get(j);
+                let cs = ks.iter().all(|k| self.steps.contains_key(k));
+                let c0 = if cs { None } else { Some(()) };
+                match (a0, b0, c0) {
```

Comment:
You could just match `cs` against true and false.

Reply:
I tried to preserve similarities between the cases. Anyways, it is a bit different now.